### PR TITLE
v5.0.0

### PR DESCRIPTION
**Pull request by flaviendelangle**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,93 @@
...
+  - The `scroll` class applied to `MuiDataGrid-columnHeaderWrapper` was renamed to `MuiDataGrid-columnHeadersInner--scrollable`
+
+- [DataGrid] `props.components.Checkbox` and `props.componentsProps.checkbox` were renamed `props.components.BaseCheckbox` and `props.componentsProps.baseCheckbox`
+
+  As a start of the effort for #3066, we renamed the slots for the checkbox to clearly indicate that it is a slot for a core component.
+
+  `components` prop change:
+  ```diff
+  <DataGrid
+    components={{
```
Comment (m4theushw) [inline CHANGELOG.md:38]: This and the following diff could be merged to reduce the size of the changelog.

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,93 @@
...
+- [core] Automatically close issues that are incomplete and inactive (#3029) @oliviertassinari
+- [core] Improve the typing of `LicenseStatus` (#3141) @Himself65
+- [core] Make `useGridColumnsPreProcessing` generic (#3092) @m4theushw
+- [core] Move column headers virtualization to hook (#3078) @m4theushw
+- [core] Move virtualization logic to hook (#3079) @m4theushw
+- [core] Rename directories to match new packages new names (#3088) @flaviendelangle
+- [core] Replace `createClientRender` with new `createRenderer` API (#3125) @flaviendelangle
+- [core] Store the event manager in a key of `GridApi` instead of making the whole `GridApi` extend it (#3069) @flaviendelangle
+- [core] Update monorepo (#3139) @m4theushw
+- [core] Use `unstable_` prefix instead of `unsafe_` for private apis (#3090) @flaviendelangle
```
Comment (m4theushw) [inline CHANGELOG.md:82]:
```suggestion
- [core] Use `unstable_` prefix instead of `unsafe_` for private APIs (#3090) @flaviendelangle
```

**Review by alexfauquette — APPROVED** [review]

**Review by DanailH — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,87 @@
...
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 5.0.0
+
+_Nov 12, 2021_
+
+🎉 This is the new stable release of the data grid component 🎉!
+
+If you want to migrate the DataGrid or DataGridPro from v4 to v5, take a look at the [migration guide](https://mui.com/components/data-grid/migration-v4/).
+This version is fully compatible with `@mui/material@5.X` and can be used with `@material-ui/core@4.x` with some additional steps as described [here](https://mui.com/components/data-grid/migration-v4/#using-mui-x-v5-with-mui-core-v4)
```
Comment (oliviertassinari) [inline CHANGELOG.md:13]: What about the release blog post :) https://mui.com/blog/mui-x-v5/? I think that it should come before everything else. Similar to https://github.com/mui-org/material-ui/releases/tag/v5.0.0.

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,87 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 5.0.0
+
+_Nov 12, 2021_
```
Comment (oliviertassinari) [inline CHANGELOG.md:8]: I assume it will happen next week.

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,87 @@
...
+  - `MuiDataGrid-columnsContainer` was renamed to `MuiDataGrid-columnHeaders`
+  - `MuiDataGrid-columnHeaderWrapper` was renamed to `MuiDataGrid-columnHeadersInner`
+  - The `scroll` class applied to `MuiDataGrid-columnHeaderWrapper` was renamed to `MuiDataGrid-columnHeadersInner--scrollable`
+
+- [DataGrid] The `props.components.Checkbox` and `props.componentsProps.checkbox` props were renamed to `props.components.BaseCheckbox` and `props.componentsProps.baseCheckbox` respectively.
+
+  As a first step for [#3066](https://github.com/mui-org/material-ui-x/issues/3066), these slots were renamed to clearly indicate that they are meant to replace a core component.
+
+  ```diff
+  <DataGrid
```
Comment (oliviertassinari) [inline CHANGELOG.md:37]: This is not the standard way git diff outputs, there is always a reserved first char for line (one of ` `, or `-`, or `+`)
```suggestion
   <DataGrid
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,6 +3,87 @@
...
+  +    baseCheckbox: {},
+    }}  
+  />;
+  ```
+
+**Note**: these changes apply to both the `DataGrid` and `DataGridPro` components.
+
+#### Changes
+
+- [DataGrid] Block multi-rows updates in `apiRef.current.updateRows` on the free plan (#3095) @flaviendelangle
```
Comment (oliviertassinari) [inline CHANGELOG.md:53]: Official name of the plan: https://mui.com/pricing/

```suggestion
- [DataGrid] Block multi-rows updates in `apiRef.current.updateRows` on the Community plan (#3095) @flaviendelangle
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -3,6 +3,87 @@
...
+- [core] Move column headers virtualization to hook (#3078) @m4theushw
+- [core] Move virtualization logic to hook (#3079) @m4theushw
+- [core] Rename directories to match new packages new names (#3088) @flaviendelangle
+- [core] Replace `createClientRender` with new `createRenderer` API (#3125) @flaviendelangle
+- [core] Store the event manager in a key of `GridApi` instead of making the whole `GridApi` extend it (#3069) @flaviendelangle
+- [core] Update monorepo (#3139) @m4theushw
+- [core] Use `unstable_` prefix instead of `unsafe_` for private APIs (#3090) @flaviendelangle
+- [core] Use official MUI repo as monorepo (#3084) @m4theushw
+
+### Docs
```
Comment (oliviertassinari) [inline CHANGELOG.md:79]:
```suggestion
### Docs

```

File: CHANGELOG.md (hunk 6 above)
Comment (oliviertassinari) [reply CHANGELOG.md:53]: @mbrookes Sometimes, I wonder if we shouldn't call it "Open source plan" like in https://www.hashicorp.com/products/vault/pricing or https://www.metabase.com/pricing/

File: CHANGELOG.md (hunk 3 above)
Comment (flaviendelangle) [reply CHANGELOG.md:13]: On monday we were saying that the blogpost would come after if I remember correctly.

Comment (oliviertassinari) [reply CHANGELOG.md:13]: > blogpost was coming after

From my perspective, the release is the blog post/published changelog. But yes, we could publish on npm today, I don't see any issue with this.

Comment (flaviendelangle) [reply CHANGELOG.md:13]: I agree that we should have started the effort on the blog post + migration guide one week earlier to release everything at once.

Comment (m4theushw) [reply CHANGELOG.md:13]: So are we going to postpone the release to after the blog post is ready? If we publish today on npm, the date in the GitHub release and the CHANGELOG won't match.

File: CHANGELOG.md [hunk 8]
```diff
@@ -3,6 +3,88 @@
...
+  +    BaseCheckbox: MyCustomCheckbox,
+    }}
+    componentsProps={{
+  -    checkbox: {},
+  +    baseCheckbox: {},
+    }}  
+  />;
+  ```
+
+**Note**: these changes apply to both the `DataGrid` and `DataGridPro` components.
```
Comment (m4theushw) [inline CHANGELOG.md:49]:
```suggestion
  **Note**: these changes apply to both the `DataGrid` and `DataGridPro` components.
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -3,6 +3,88 @@
...
+- [core] Use official MUI repo as monorepo (#3084) @m4theushw
+
+### Docs
+
+- [docs] Fix broken example in the component slot example (#3123) @Himself65
+- [docs] Fix inline previews (#3081) @DanailH
+- [docs] Fix the client-side validation link clarity (#3100) @oliviertassinari
+- [docs] Improve `rowCount` CSS class description (#3072) @ZeeshanTamboli
+- [docs] Use core repo constants for doc internationalization (#3143) @flaviendelangle
+- [core] Adapt changelog script to GitHub DOM modification (#3087) @alexfauquette
```
Comment (m4theushw) [inline CHANGELOG.md:86]: This one should not be here if it's "core".  https://github.com/mui-org/material-ui-x/pull/3160#discussion_r747500087

File: CHANGELOG.md (hunk 3 above)
Comment (flaviendelangle) [reply CHANGELOG.md:13]: I don't think we should postpone it.

Comment (DanailH) [reply CHANGELOG.md:13]: Actually not much is left from the blogpost - we can add the migration to it and add a `whats next` content. We can drop the 'Features highlights' section

**Pull request merged**